src/com/haskforce/internal/HaskForceInternal.scala

```diff
@@ -0,0 +1,20 @@
+package com.haskforce.internal
+
+import com.haskforce.codeInsight.HaskellCompletionCacheLoader
+import com.intellij.ide.ApplicationInitializedListener
+import com.intellij.openapi.application.{Application, ApplicationManager}
+import com.intellij.openapi.project.ProjectManager
+import com.intellij.util.messages.Topic
+
+class HaskForceInternal extends ApplicationInitializedListener {
+  override def componentsInitialized(): Unit = {
+    val app = ApplicationManager.getApplication()
+    registerTopic(app, ProjectManager.TOPIC, new HaskellCompletionCacheLoader)
+  }
+
+  def registerTopic[A](app: Application, topic: Topic[A], instance: A): Unit = {
+    app.getMessageBus.connect(app).subscribe(topic, instance)
+  }
+}
+
+
```

Comment:
Seems like this might need to be registered in the plugin.xml as well. Also, I think this might be better named as `com.haskforce.ide.HaskForceInitializedListener` to match the interface. Plus, `internal` seems a little vague for what this class is used for.

Reply:
we can rename it to whatever makes sense to you